hastory-server/src/Data/Hastory/Server/Handler/Users.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Handler.Users where
+
+import Data.Hastory.Server.Handler.Import
+
+createUserHandler :: UserForm -> HastoryHandler UserId
+createUserHandler userForm@UserForm {..} = do
+  mUser <- runDB . getBy $ UniqueUsername userFormUserName
+  if isNothing mUser && isValid userForm
+    then buildAndInsertUser userFormUserName userFormPassword
+    else throwError err400
```

Comment:
if the `userForm` is invalid, we want to show the reason, no? you can use `prettyValidate` and case-match

Reply:
Done.